controllers/solrcloud_controller.go

```diff
@@ -699,6 +695,41 @@ func reconcileCloudStatus(r *SolrCloudReconciler, solrCloud *solr.SolrCloud, log
 	return outOfDatePods, outOfDatePodsNotStarted, availableUpdatedPodCount, nil
 }
 
+func isPodReadyForBackup(pod corev1.Pod, backupOptions *solr.SolrBackupRestoreOptions) bool {
```

Comment:
As written, this method checks if a single pod is ready to handle a backup for any/all of its repositories.  This information gets surfaced to consumers under the `backupRestoreReady` boolean field of `SolrCloudStatus`.

We can definitely change it to take a specific repository, but since this call isn't happening at backup-time, we'd still have to call it in a loop for each repository defined on the solrcloud instance and then "AND" each of those results together into the single `backupRestoreReady` value.  So I'm not really sure that gets us anything?

Unless you're thinking that we'd change SolrCloudStatus to include a distinct "backupReadiness" status for each repository so that we could check the one we care about at "backup time".  I'd be OK with that if that's what you're suggesting, just wanted to double check we're envisioning the same thing here.

Reply:
Yeah I was thinking that the SolrCloud status could start providing a list of ready backup Repos. If you don't think this is very feasible that's fine. We can always add it later.